test/src/test/java/org/corfudb/runtime/LogReplicationLogicalGroupClientShimTest.java

```diff
@@ -0,0 +1,330 @@
+    }
+    private static String namespace;
+    private static String registrationTableName;
+    private static String metadataTableName;
+    private static String clientName;
+    private static CorfuRuntime runtime;
+    private static CorfuStoreShim store;
+
+    @Before
+    public void loadProperties() throws Exception {
+        namespace = "CorfuSystem";
+        registrationTableName = "LogReplicationRegistrationTable";
+        metadataTableName = "LogReplicationModelMetadataTable";
+        clientName = "client";
+
+        runtime = getTestRuntime();
+        store = new CorfuStoreShim(runtime);
+    }
+
+    /**
+     * Test registering replication client
+     *
+     */
+    @Test
+    public void testRegisterReplicationClient() throws InvocationTargetException, NoSuchMethodException, IllegalAccessException {
+        final Table<ClientRegistrationId, ClientRegistrationInfo, ManagedResources> table = store.openTable(
+                namespace,
+                registrationTableName,
+                ClientRegistrationId.class,
+                ClientRegistrationInfo.class,
+                ManagedResources.class,
+                TableOptions.fromProtoSchema(Uuid.class));
+
+        // Test registering client with null/empty client name
+        Assert.assertThrows(IllegalArgumentException.class, () ->  new LogReplicationLogicalGroupClient(runtime, null));
+        Assert.assertThrows(IllegalArgumentException.class, () ->  new LogReplicationLogicalGroupClient(runtime, ""));
+
+        // Test registering a new client
+        new LogReplicationLogicalGroupClient(runtime, clientName);
+        final int expectedNumberRegisteredClients = 1;
+        Assert.assertEquals(expectedNumberRegisteredClients, table.count());
+
+        // Test registering a duplicate client
+        new LogReplicationLogicalGroupClient(runtime, clientName);
+        final int expectedNumberRegisteredClients1 = 1;
+        Assert.assertEquals(expectedNumberRegisteredClients1, table.count());
+
+        // Test registering 2 additional clients
+        new LogReplicationLogicalGroupClient(runtime, "client1");
+        new LogReplicationLogicalGroupClient(runtime, "client2");
+        final int expectedNumberRegisteredClients2 = 3;
+        Assert.assertEquals(expectedNumberRegisteredClients2, table.count());
+    }
+
+    /**
+     * Test add destination
+     *
+     * @throws Exception exception
+     */
+    @Test
+    public void testAddDestination() throws Exception {
+        final Table<ClientDestinationInfoKey, DestinationInfoVal, ManagedResources> table = store.openTable(
+                namespace,
+                metadataTableName,
+                ClientDestinationInfoKey.class,
+                DestinationInfoVal.class,
+                ManagedResources.class,
+                TableOptions.fromProtoSchema(Uuid.class));
+
+        LogReplicationLogicalGroupClient client = new LogReplicationLogicalGroupClient(runtime, clientName);
```

Comment:
nit: we can open the 2 tables, register and the sourceMetadata table, and invoke `LogReplicationLogicalGroupClient` in the `@Before` function.